src/token/impl/src/canister.rs

```diff
@@ -138,7 +137,10 @@ impl Auction for TokenCanister {
     }
 
     fn disburse_rewards(&self) -> Result<AuctionInfo, AuctionError> {
-        token_api::canister::is20_auction::disburse_rewards(self)
+        token_api::canister::is20_auction::disburse_rewards(
+            &mut self.state().borrow_mut(),
```

Comment:
Why that complicated? What happened to the previous version of `disburse_rewards`?

Reply:
There are a few reasons why we don't want to pass the canister itself into the inner business logic of the canister:
1. The mutability guards that are broken by the canister. Since the state in the canister is hidden behind `RefCell`, while we work with the canister object itself we loose this mutability checks. If we only work with state itself, we can be sure that mutable objects are mutable and immutable objects are immutable.
2. Usually it's a good practice to receive in a function minimum required data for the function to function. In this case the function only needs the token state to operate, so there's no reason to give it auction state, metrics state, etc.